File: STRIVER/video/vid-5/pattern10/p10.py
# *
# * *
# * * *
# * * * * 
# * * * * *
# * * * * 
# * * *
# * *
# *


def p10(row:int)->None:
    # upper
    for i in range(row):
        for j in range(i):
            print("* ",end="")
        print("")

    # #lower 
    for i in range(row, -1, -1):
        for j in range(i):
            print("* ",end="")
        print("")

row = 4
p10(row)

File: STRIVER/video/vid-5/pattern10/p10.cpp
// *
// * *
// * * *
// * * * * 
// * * * * *
// * * * * 
// * * *
// * *
// *

#include <iostream>
using namespace std;

int main()
{
    int row = 4;

    // upper
    for (int i = 0; i < row; i++)
    {
        for (int j = 0; j < i; j++)
        {
            cout << "* ";
        }
        cout << endl;
    }

    // lower
    for (int i = row; i >= 0; i--)
    {
        for (int j = 0; j < i; j++)
        {
            cout << "* ";
        }
        cout << endl;
    }
    return 0;
}